app/services/ErrorTranslationService.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import models.errors.FunctionalError
+import models.errors.XmlError
+import models.values.ErrorPointer
+import models.values.ErrorType
+
+import javax.inject.Inject
+import javax.inject.Singleton
+import com.google.inject.ImplementedBy
+
+@ImplementedBy(classOf[ErrorTranslationServiceImpl])
+trait ErrorTranslationService {
+  def translateErrorPointer(pointer: ErrorPointer): String
+  def translateFunctionalError(error: FunctionalError): String
+  def translateXmlError(error: XmlError): String
+}
+
+@Singleton
+class ErrorTranslationServiceImpl @Inject() extends ErrorTranslationService {
+  val AccessCodePointer        = ErrorPointer("GRR(1).ACC(1).Access code")
```

Comment:
Will there ever be more than one GRR returned in an error?

Reply:
No, you can only query one guarantee reference at a time using our API, but technically the message structure allows 99 GRNs to be queried - we just don't allow users to do that